staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -417,7 +417,8 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			// It doesn't make sense to re-list all objects because most likely we will be able to restart
 			// watch where we ended.
 			// If that's the case begin exponentially backing off and resend watch request.
-			if utilnet.IsConnectionRefused(err) {
+			// Do the same for "429" errors.
+			if utilnet.IsConnectionRefused(err) || apierrors.IsTooManyRequests(err) {
```

Comment:
what would old clients do in response to a 429 if they miss this block and returned the error below?

Reply:
Yes - they would just return an error and relist.

Reply:
What I can also imagine - is just the client-part now (I think those will be useful eventually anyway).
[Whether this is cherrypickable or not is a separate discussion.]

And start utilizing this in the server some time in the future.